lib/active_campaign/v2/clients/contact.rb

```diff
@@ -60,6 +60,27 @@ def contact_list(ids: 'all', **args)
           get('/admin/api.php', query: args.merge(api_action: 'contact_list', ids: Array(ids).join(',')))
         end
 
+        # This can be used to search for contacts based on a custom field setup in ActiveCampaign.
+        # It will return a JSON object as the response with the following key structure:
+        #
+        # {
+        #   '0': DATA FOR FIRST RESULT,
+        #   '1': DATA FOR SECOND RESULT,
+        #   'result_code': 1,
+        #   'result_message': 'Success: Something is returned',
+        #   'result_output': 'json',
+        # }
+        #
+        # NOTE: This is most useful for looking up a contact with a unique identifier stored in a
+        # custom field. Since the results are just stored in the JSON objects with a cardinal index,
+        # it would be tricky to navigate this response with many results. That being said, if you
+        # expect there to be one and only one result with a unique identifier, it should work.
+        #
+        # NOTE: Contacts that are not subscribed to any list will not be returned in the result set.
+        def custom_field_search(field:, value:, **args)
```

Comment:
Would it be helpful to do `field.upcase` or is it reasonable to assume that consumers of this API know that custom fields are ALL_UPCASE?

Reply:
Great call here. Personally, I'd prefer to pass a symbol, so I included a `to_s` as well to handle it. Now users can submit:

```
:test_field
'test_field'
'TEST_FIELD'
```

And they will be formatted appropriately.